cluster-cm/src/main/java/com/sequenceiq/cloudbreak/cm/ClouderaManagerDeployClientConfigProvider.java

```diff
@@ -0,0 +1,240 @@
+package com.sequenceiq.cloudbreak.cm;
+
+import java.io.IOException;
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import org.joda.time.DateTime;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+
+import com.cloudera.api.swagger.ClustersResourceApi;
+import com.cloudera.api.swagger.client.ApiClient;
+import com.cloudera.api.swagger.client.ApiException;
+import com.cloudera.api.swagger.client.ApiResponse;
+import com.cloudera.api.swagger.client.Pair;
+import com.cloudera.api.swagger.model.ApiCommand;
+import com.cloudera.api.swagger.model.ApiCommandList;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Splitter;
+import com.sequenceiq.cloudbreak.cm.model.CommandResource;
+import com.sequenceiq.cloudbreak.domain.stack.Stack;
+import com.sequenceiq.cloudbreak.service.CloudbreakException;
+import com.squareup.okhttp.HttpUrl;
+import com.squareup.okhttp.OkHttpClient;
+import com.squareup.okhttp.Request;
+import com.squareup.okhttp.Response;
+
+@Service
+public class ClouderaManagerDeployClientConfigProvider {
+
+    private static final Logger LOGGER =
+            LoggerFactory.getLogger(ClouderaManagerDeployClientConfigProvider.class);
+
+    private static final String CMF_COMMANDS_COMMAND_TABLE_URI_PATH = "/cmf/commands/commandTable";
+
+    private static final String DEPLOY_CLIENT_CONFIG_COMMAND_NAME = "DeployClusterClientConfig";
+
+    private static final int INTERVAL_MINUTES = 30;
+
+    private static final int ERROR_CODES_FROM = 400;
+
+    private final Integer interruptTimeoutSeconds;
+
+    public ClouderaManagerDeployClientConfigProvider(
+            @Value("${cb.cm.client.commands.deployClientConfig.interrupt.timeout.seconds:}") Integer interruptTimeoutSeconds) {
+        this.interruptTimeoutSeconds = interruptTimeoutSeconds;
+    }
+
+    public BigDecimal calculateDeployClientConfigCommandId(
```

Comment:
calculate is a bit misleading, I would rename it to something like deployClientConfigAndGetCommandId

Reply:
can be getCommandId as well as that's clear that is for deployClientConfig from the name (of course it does not tell it uses 3 different way to do it)